### ISSUE TITLE
Unable to save tensorflow /keras model using R in Rstudio on Google Cloud Platform

### ISSUE THREAD

**Issue by ashishsr123** [rstudio/keras#783]
I am using Rstudio on Google cloud platform and want to deploy a trained model but facing error. I am using examples present at https://tensorflow.rstudio.com/keras/

I am unable to export model to google cloud storage , after which I can create model version on google cloud storage and use if for predictions. I am able to do predictions in Rstudio.

library(keras)

FLAGS <- flags(flag_numeric("dropout_rate", 0.4))

mnist <- dataset_mnist()
x_train <- mnist$train$x
y_train <- mnist$train$y
x_test <- mnist$test$x
y_test <- mnist$test$y

x_train <- array_reshape(x_train, c(nrow(x_train), 784))
x_test <- array_reshape(x_test, c(nrow(x_test), 784))
x_train <- x_train / 255
x_test <- x_test / 255

y_train <- to_categorical(y_train, 10)
y_test <- to_categorical(y_test, 10)

model <- keras_model_sequential()

model %>%
  layer_dense(units = 256, activation = 'relu', input_shape = c(784)) %>%
  layer_dropout(rate = FLAGS$dropout_rate) %>%
  layer_dense(units = 128, activation = 'relu') %>%
  layer_dropout(rate = 0.3) %>%
  layer_dense(units = 10, activation = 'softmax')

model %>% compile(
  loss = 'categorical_crossentropy',
  optimizer = optimizer_rmsprop(),
  metrics = c('accuracy')
)

model %>% fit(
  x_train, y_train,
  epochs = 20, batch_size = 128,
  validation_split = 0.2
)

model %>% evaluate(x_test, y_test)
model %>% predict_classes(x_test)

# everything works fine till above line 

export_savedmodel(model, "savedmodel") # Error line

Error message: for line in Rstudio --> export_savedmodel(model, "savedmodel")

Error in export_savedmodel.keras.engine.training.Model(model, "savedmodel") : 
      'export_savedmodel()' is currently unsupported under the TensorFlow Keras implementation, consider using 'tfestimators::keras_model_to_estimator()'

After this I tried below code:

#export_savedmodel(model, "savedmodel") <-- replace this line with below code
tfe_model <- tfestimators::keras_model_to_estimator(model)
export_savedmodel(tfe_model, "savedmodel") 

Error message:
Error in export_savedmodel.tf_estimator(tfe_model, "savedmodel") : Currently only classifier and regressor are supported. Please specify a custom serving_input_receiver_fn.

Thanks
Ashishsr123

**Comment by dfalbel** [comment]
We are working on a fix for this issue. In the meantime can you share your `tf_config()`, please?

**Comment by ashishsr123** [comment]
I have 2 different models and giving config for both.
For the model mentioned in the issue above:

> tf_config()
TensorFlow v1.13.1 (~/.virtualenvs/r-tensorflow/local/lib/python2.7/site-packages/tensorflow)
Python v2.7 (~/.virtualenvs/r-tensorflow/bin/python)

for other model its also failing
model details at below location: 
https://github.com/rstudio/rstudio-conf/blob/master/2018/Deploying_TensorFlow_Models--Javier%20Luraschi/deploy-tensorflow-models-walkthrough.Rmd
> tf_config()
TensorFlow v1.9.0 (~/.virtualenvs/r-tensorflow/local/lib/python2.7/site-packages/tensorflow)
Python v2.7 (~/.virtualenvs/r-tensorflow/bin/python)

**Comment by dfalbel** [comment]
@ashishsr123 It should work now with the dev version of keras. Please install it with `devtools::install_github("rstudio/keras")`. 

Feel free to reopen this issue if the problem is not solved.

**Comment by ashishsr123** [comment]
I still face issue.

I install dev version of keras and ran code posted in the issue.
in the very last I ran below code and got error.

> export_savedmodel(model, "savedmodel")
Error in export_savedmodel.keras.engine.training.Model(model, "savedmodel") : 
  'export_savedmodel()' is currently unsupported under the TensorFlow Keras implementation, consider using 'tfestimators::keras_model_to_estimator()'.

additionally I tried below code .
tfe_model <- tfestimators::keras_model_to_estimator(model)
export_savedmodel(tfe_model, "savedmodel")
error message.
Error in export_savedmodel.tf_estimator(tfe_model, "savedmodel") : 
  Currently only classifier and regressor are supported. Please specify a custom serving_input_receiver_fn.

**Comment by ashishsr123** [comment]
can you please open issue again  ?
 I am not able to re-open closed issue.

**Comment by dfalbel** [comment]
Would you update your TensorFlow installation? You could do that with `install_tensorflow()`.

**Comment by dfalbel** [comment]
The following message

> `'export_savedmodel()' is currently unsupported under the TensorFlow Keras implementation, consider using 'tfestimators::keras_model_to_estimator()'.`

does not exist anymore in the dev version of Keras. This indicates that you don't have the gitthub version of keras installed. Try installing it with `devtools::install_github("rstudio/keras")`.

**Comment by ashishsr123** [comment]
thanks I was able to export_savedmodel()
